packages/ui-components/src/components/tag/tag.tsx

```diff
@@ -0,0 +1,22 @@
+import { Component, Host, Prop, h } from '@stencil/core';
+
+@Component({
+	tag: 'kv-tag',
+	styleUrl: 'tag.scss',
+	shadow: true
+})
+export class KvTag {
+	/** (optional) Tag text */
+	@Prop({ reflect: true }) text?: string;
+
+	render() {
+		return (
+			<Host>
+				<div class="tag-container">
+					<slot name="left-slot"></slot>
+					<div class="tag-text">{this.text}</div>
```

Comment:
if text is optional 
```
{ this.text && (<div ... }

```

Reply:
After a rebase with dev you already have a `string.helper.ts` inside `utils` with `isValidLabel` that validate if string is empty or only contain empty spaces. You should use here for validation 
```
{ isValidLabel(this.text) && ... }

```